src/services/recentlyPlayedService.tsx

```diff
@@ -0,0 +1,117 @@
+import type { AudioFeaturesResponse } from "../types/spotify-web-api";
+import getMultiTrackAudioFeatures from "./multiTrackAudioFeaturesService";
+import type { HistoricalMetrics } from "../types/enhancify";
+
+async function getRecentlyPlayedTracksMetrics() : Promise<HistoricalMetrics | {}> {
+    
+    // Set the API option
+    let apiOptions = {
+                        limit: "50",
+                        after: "",
+                     };
+    // Get the date of the first day of the week
+    let today = new Date();
+    let day = today.getDay() || 7;
+    if (day != 1) {
+        today.setHours(-24 * (day - 1));
+    }
+    // Convert the date to UNIX format
+    apiOptions.after = (today.getTime() / 1000).toFixed(0);
```

Comment:
Could you explain this part of the code? What's the calculation here and from what time range are we requesting the 50 songs?